cosmo_tester/framework/cfy_helper.py

```diff
@@ -97,6 +101,37 @@ def bootstrap(self,
                 task_retry_interval=task_retry_interval,
                 verbose=verbose,
                 debug=debug).wait()
+            self.upload_plugins(cfy)
+
+    def _download_wagons(self):
+        self.logger.info('Downloading Wagons...')
+
+        wagon_paths = []
+
+        plugin_urls_location = os.environ.get(
+            'PLUGIN_URLS_LOCATION',
+            'https://raw.githubusercontent.com/'
+            'cloudify-cosmo/cloudify-versions/master/plugin-urls.yaml',
+        )
+
+        plugins = yaml.load(
+            requests.get(plugin_urls_location).text
+        )['plugins']
+        for plugin in plugins:
+            self.logger.info(
+                'Downloading: {0}...'.format(plugin['wgn_url'])
+            )
+            wagon_paths.append(
+                download_file(plugin['wgn_url'], keep_existing=True)
```

Comment:
you probably want to specify some destination if you want the caching logic to take place